include/deal.II/lac/linear_operator.h

```diff
@@ -104,25 +166,51 @@ null_operator(const LinearOperator<Range, Domain> &);
  * this object to encapsulate matrix object of medium to large size (as a rule
  * of thumb, sparse matrices with a size $1000\times1000$, or larger).
  *
+ * @note In order to use Trilinos or PETSc sparse matrices and preconditioners
+ * in conjunction with the LinearOperator class, it is necessary to extend the
```

Comment:
What's also unclear is what kind of *interface* these classes have to have. Would that be worth describing as well? I.e., what does the `LinearOperator` class expect template arguments for the `Payload` type to provide?